### DISCUSSION TITLE
Custom name service integration

### DISCUSSION THREAD

**Discussion by romanOogabooga** [rainbow-me/rainbowkit#2196] [Ideas]
Hey guys, I'm struggling to integrate Rainbowkit with custom name service. I'm working on Berachain and need to integrate their name service. They have contract an I need to call function that will return the name for address.

In Rainbowkit
We have Account object where we have diplayName field. It's easy to change it in Custom Connect Button component, but looks like it's impossible to change it for all Rainbow kit. I want to change it in Modal. Does smb know how to solve it?

**Comment by DanielSinclair** [comment]
@romanOogabooga For Berachain, is the name service resolvable by the ENS resolver? We rely on Wagmi and Viem for this; if it is compatible with ENS, it may just require a Viem change.

**Reply by romanOogabooga** [reply]
no, for now unfortunately I did it manually because they just have a contract with names
